src/controllers/studyplans.controller.ts

```diff
@@ -1,22 +1,32 @@
+import { Request, Response } from "express";
 import { convertResultToStudyPlan } from "@utils/converter";
 import { getConnection } from "@utils/mongoconnection";
-import { Request, Response } from "express";
+import { getRedisClient } from "@utils/redisclient";
+
+const cacheTTL = 60 * 24 * 30; // cache for 30 days
 
-// to be removed after frontend has migrated
```

Comment:
This file is deprecated, please make changes to `resources.controllers.ts` instead

Reply:
oh hai hai, i'll try to redo it with all the parts you mentioned